buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -262,45 +262,47 @@ private void setDistributionType(ElasticsearchDistribution distribution, TestDis
         }
     }
 
-    @Override
-    public void plugin(RegularFileProperty plugin) {
-        this.plugins.add(plugin.map(p -> p.getAsFile().toURI()));
+    // package protected so only TestClustersAware can access
+    @Internal
+    List<Configuration> getPluginAndModuleConfigurations() {
+        return pluginAndModuleConfigurations;
     }
 
-    @Override
-    public void plugin(Provider<URI> plugin) {
-        requireNonNull(plugin, "Plugin name can't be null");
-        checkFrozen();
-        if (plugins.contains(plugin)) {
-            throw new TestClustersException("Plugin already configured for installation " + plugin);
+    // creates a configuration to depend on the given plugin project, then wraps that configuration
+    // to grab the zip as a file provider
+    private Provider<RegularFile> maybeCreatePluginOrModuleDependency(String path) {
+        Project depProject = project.project(path);
+        Configuration configuration = project.getConfigurations().findByName(depProject.getName());
+        if (configuration == null) {
+            configuration = project.getConfigurations().create(depProject.getName());
+            DependencyHandler deps = project.getDependencies();
+            deps.add(depProject.getName(), deps.project(Map.of("path", path, "configuration", "zip")));
         }
-        this.plugins.add(plugin);
+        pluginAndModuleConfigurations.add(configuration);
+        Provider<File> fileProvider = configuration.getElements().map(s -> s.stream().findFirst().get().getAsFile());
```

Comment:
Let's use `orElseThrow()` instead of `get()` and throw an exception with a useful message rather than the default `NoSuchElementException` that would result here if something went wrong.